### ISSUE TITLE
build reports all green checkmarks, fires after_success, shows as failed on the list

### ISSUE THREAD

**Issue by easel** [Shippable/support#874]
Not much to say other than the subject. The build is red, but all the steps passed. How can I troubleshoot what's going on?

**Comment by rageshkrishna** [comment]
@easel Sorry for taking so long to get to this. I've taken a look at some of your more recent builds and I can't seem to find any that are running in to the issues described here. I see a couple of failed/error tests but I'm not sure that they're related to the dependency issue in your previous comment. Can you please let me know if your most recent builds are behaving as they should?

I also see that one of your builds is running on the shippableimages/ubuntu1x04_scala image and trying to launch xvfb. Our language specific images don't come with any pre-installed services  so if you need xvfb, you're going to have to install it yourself as part of the `before_install` step in your YML file.

**Comment by easel** [comment]
@rageshkrishna thanks for the details, and my own apologies for being slow to respond. I've added an apt-get install -y xvfb to my shippable.yml, and now I'm getting "missing host" type errors. I'm assuming that's transient, but perhaps not...

I'll go ahead and strip our the XVFB stuff for the time being and see where that leaves me.

**Comment by easel** [comment]
The specific error:

 apt-get -y install xvfb
Reading package lists...
Building dependency tree...
Reading state information...
The following extra packages will be installed:
  libxkbfile1 x11-xkb-utils xfonts-base xserver-common
Suggested packages:
  xfs xserver
The following NEW packages will be installed:
  libxkbfile1 x11-xkb-utils xfonts-base xserver-common xvfb
0 upgraded, 5 newly installed, 0 to remove and 18 not upgraded.
Need to get 7,181 kB of archives.
After this operation, 13.2 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu/ trusty/main libxkbfile1 amd64 1:1.0.8-1 [74.2 kB]
Get:2 http://archive.ubuntu.com/ubuntu/ trusty/main x11-xkb-utils amd64 7.7+1 [156 kB]
Get:3 http://archive.ubuntu.com/ubuntu/ trusty/main xfonts-base all 1:1.0.3 [6,180 kB]
Err http://archive.ubuntu.com/ubuntu/ trusty-updates/main xserver-common all 2:1.15.1-0ubuntu2.1
  404  Not Found [IP: 91.189.91.15 80]
Err http://archive.ubuntu.com/ubuntu/ trusty-updates/main xvfb amd64 2:1.15.1-0ubuntu2.1
  404  Not Found [IP: 91.189.91.15 80]
Fetched 6,409 kB in 0s (12.2 MB/s)
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/x/xorg-server/xserver-common_1.15.1-0ubuntu2.1_all.deb  404  Not Found [IP: 91.189.91.15 80]

E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/x/xorg-server/xvfb_1.15.1-0ubuntu2.1_amd64.deb  404  Not Found [IP: 91.189.91.15 80]

E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?

**Comment by pmoust** [comment]
@easel run `apt-get update` first.

**Comment by easel** [comment]
Did that, it worked. Now it's refusing to start xvfb from /etc/init.d -- I'm going to try xvfb-run instead.

/etc/init.d/xvfb start
/home/shippable/build/build.sh: line 102: /etc/init.d/xvfb: No such file or directory

**Comment by easel** [comment]
Ok, now I'm back to the previous sbt update issues, I'm assuming you guys have a proxy or something configured in ~/.sbt/repositories?

[warn] 	Note: Unresolved dependencies path:
[warn] 		com.typesafe.play:play-ws_2.11:2.3.7 (/root/workspace/src/github.com/telepathdata/telepath-play/src/scenarios/build.sbt#L11-16)
[warn] 		  +- com.telepathdata:scenarios_2.11:0.17.0.20150116.6cee53
[warn] 		com.typesafe.play:play-test_2.11:2.3.7 (/root/workspace/src/github.com/telepathdata/telepath-play/src/scenarios/build.sbt#L11-16)

I've got a local sbt launcher script I'm going to try, I've had issues with older issues of sbt previously.

**Comment by easel** [comment]
Ok, I've gotten all my issues worked out now.  I *think* I'm building reliably, functional tests work with xvfb, test status and coverage are being properly identified, etc. A few notes, take them as you will:

* I had some intermittent problems with caching. With it on, I still seemed to be re-downloading ~/.sbt and ~/.ivy2, but I ran into some compile errors that indicate parts of the environment were sticking around. It would be helpful to have a better idea of exactly what's happening behind the scenes when caching is enabled, and perhaps to be able to configure what folders get picked up and what are ignored.

* I had to apt-get install a bunch of stuff by hand, and to run apt-get update first. The documentation said to just enable the xvfb service, although perhaps I looked in the wrong place. Perhaps it would be good to differentiate between "services" and rolling your own. For me, apt-get install xvfb and then xvfb-run worked well rather than using the service.

* I experienced a number of issues with respect to sbt bootstrapping. I'm not sure if the version coming in ubuntu is old, or if there's a proxy location issue, or something else, but in order to finally rectify them I introduced my own sbt wrapper script and repository configuration that force EVERYTHING to be downloaded exactly as a specify. This is a common issue with sbt, unfortunately.

* Builds still seem to be fairly slow. I'm seeing 30 minutes to do what my mac laptop does in ~5.
